examples/reshare/src/main.rs

```diff
@@ -1,22 +1,73 @@
 #![doc = include_str!("../README.md")]
 
-use crate::application::{EdScheme, ThresholdScheme};
+use crate::{
+    application::{EdScheme, ThresholdScheme},
+    dkg::{ContinueOnUpdate, PostUpdate, Update, UpdateCallBack},
+    setup::ParticipantConfig,
+};
 use clap::{Args, Parser, Subcommand};
-use commonware_cryptography::bls12381::primitives::variant::MinSig;
+use commonware_codec::Encode;
+use commonware_cryptography::{bls12381::primitives::variant::MinSig, ed25519::PublicKey};
 use commonware_runtime::{
     tokio::{self, telemetry::Logging},
     Metrics, Runner,
 };
-use std::path::PathBuf;
-use tracing::Level;
+use commonware_utils::hex;
+use std::{future::Future, path::PathBuf, pin::Pin};
+use tracing::{info, Level};
 
 mod application;
 mod dkg;
 mod engine;
+mod namespace;
 mod orchestrator;
+mod self_channel;
 mod setup;
 mod validator;
 
+struct SaveFileOnUpdate {
```

Comment:
nit: can you add some comments explaining what this is for (and why we don't use something like Metadata)?